File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        int[] result = new int[nums.length];

        int prev = 1;
        for(int i = 0; i < nums.length; i++){
            result[i] = prev;
            prev = nums[i] * prev;
        }

        prev = 1;
        for(int i = nums.length-1; i >= 0; i--){
            result[i] = prev * result[i];
            prev = nums[i] * prev;
        }        

        return result;        
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        result = [] * len(nums)
        prev = 1

        for i in range(len(nums)):
            result.append(prev)
            prev = nums[i] * prev   

        prev = 1

        for i in range(len(nums)-1, -1, -1):
            result[i] = prev * result[i]
            prev = nums[i] * prev

        return result        
